### PR TITLE
Display rules.abaplint.org for check and individual messages

### PR DESCRIPTION
**Pull request by mbtools**
Clicking on "i" icon for abaplint root and categories will display "rules.abaplint.org" in external browser.

Clicking on "i" icon for individual messages will display "rules.abaplint.org/<rule>" in a popup:

image

The display of these abaplint webpages in the browser popup has a couple issue:
1. alignment of top icons (minor)
2. missing display of default config for rule (might be a simple change to add the domain to the required scripts; not sure)

These items would have to be addressed in the generation of the rules website.

Currently, the mapping of rules ids to abap message codes is hardcoded in the class_constructor of zcl_abaplint_result. Ideally, this could be retrieved from the abaplint metadata automatically. Until then, it will be necessary to maintain the mapping if new rules are created.

### CODE REVIEW

**Comment by mbtools** [comment]
@larshp how to fix these lint errors? 

can we test adding https://rules.abaplint.org domain here?
https://github.com/abaplint/abaplint/blob/master/web/rules/src/rule_page.ts#L28:L29

**Comment by larshp** [comment]
I've added the class to [deps](https://github.com/abaplint/deps), https://github.com/abaplint/deps/commit/957b514b617a242bceea51458612e403bd295519

File: src/zcl_abaplint_result.clas.abap [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+
+  METHOD class_constructor.
+
+    DATA:
+      lv_map TYPE string,
+      lt_map TYPE TABLE OF string,
+      ls_map TYPE ty_map.
+
+    lv_map = '7bit_ascii,LINT_0001|abapdoc,LINT_0002|allowed_object_naming,LINT_0003|allowed_object_types,LINT_0004|'
```
Comment (larshp) [inline src/zcl_abaplint_result.clas.abap:50]: hmm, how to keep this updated? rules are added and removed every month

Comment (larshp) [reply src/zcl_abaplint_result.clas.abap:50]: perhaps add the hashes, so https://rules.abaplint.org/12abhash123/ works

**Comment by mbtools** [comment]
I thought we could add one http call to get back a list of all rules. But I think we have that already with getting the default json which I could parse.

**Comment by larshp** [comment]
and it will only be called when the user presses the "i" icon?

**Comment by mbtools** [comment]
The class constructor will be called once per SCI transaction. So there would be one http call to get the default json. 

The lint_xxxxx message numbers would not be stable as new rules are added, though. If we want that, then we need a way to get this mapping from the server. Could be a field in the meta for each rule that’s then included in default json.

**Comment by larshp** [comment]
list_rules endpoint added, https://github.com/abaplint/abaplint-sci-server/pull/219

**Comment by mbtools** [comment]
popup idea doesn't work since it won't load monaco. starting over with new PR

**Pull request closed without merging**